### PR TITLE
Add a doc to help troubleshoot package mixups

### LINKED ISSUE #18347: Add instructions to diagnose/fix errors caused by mixing Microsoft and source-built Linux packages
Now that source-built .NET Core is becoming more widespread (Fedora 32 and CentOS 8 semi recently), people are hitting issues when some packages on their machines come from the Microsoft repos and others from the distro archive. They don't mix well: they have different dotnet roots, `/usr/share/dotnet` vs. `/usr/lib64/dotnet`.

Instances:

* https://github.com/dotnet/core/issues/4605#issuecomment-618524478
* https://github.com/dotnet/core/issues/4655
* https://github.com/dotnet/installer/issues/9254
* https://github.com/dotnet/core/issues/5746
* https://github.com/dotnet/sdk/issues/15476
* https://github.com/dotnet/core/issues/5018#issuecomment-757400170
* https://bugzilla.redhat.com/show_bug.cgi?id=1847206
* https://stackoverflow.com/q/65422998
* https://old.reddit.com/r/Fedora/comments/kspz8i/troubles_with_installing_net
* https://github.com/dotnet/sdk/issues/15785
* https://github.com/dotnet/sdk/issues/15863

Related (how to mix source-built 3.1 with Msft 2.1, intentionally):

* https://github.com/dotnet/docs/pull/18070#issuecomment-620680909

Related (tracking non-doc fixes/mitigations):

* https://github.com/dotnet/runtime/issues/47500

Our docs should probably help people get past these problems since we're maintaining instructions for both Microsoft and source-built .NET Core. (I believe the direction will usually if not always be from Microsoft => source-build.)

/cc @omajid @NikolaMilosavljevic @dleeapho @MichaelSimons @leecow @Thraka 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c14a3e31-58b4-557e-20d8-57c81ed95296
* Version Independent ID: 9a7a2a48-910f-17ce-5388-27f068f427f9
* Content: [Install .NET Core on Fedora 32 - package manager - .NET Core](https://docs.microsoft.com/en-us/dotnet/core/install/linux-package-manager-fedora32)
* Content Source: [docs/core/install/linux-package-manager-fedora32.md](https://github.com/dotnet/docs/blob/master/docs/core/install/linux-package-manager-fedora32.md)
* Product: **dotnet-core**
* GitHub Login: @Thraka
* Microsoft Alias: **adegeo**

### PR DESCRIPTION
**Pull request by omajid**
Now that source-built .NET Core is becoming more widespread, people are hitting issues when some packages on their machines come from the Microsoft repos and others from the distro archive. They don't mix well: they have different dotnet roots, /usr/share/dotnet vs. /usr/lib64/dotnet.

Document some ways to identify this problem as well as some solutions based on what the user's goal is.

Fixes: #18347

cc @dagood @NikolaMilosavljevic @dleeapho @MichaelSimons @leecow @adegeo

### CODE REVIEW

**Comment by adegeo** [comment]
@omajid Thank you for writing this content! I'll do an edit pass on it.

**Comment by adegeo** [comment]
@omajid Review [this commit diff](https://github.com/dotnet/docs/pull/24206/commits/57ee911ea59f9c842496fb865451f085ff4e6bf9) it shows how I changed your article. I want to make sure I didn't lose any info.

**Comment by omajid** [comment]
Thanks, that is a great edit!

File: docs/core/install/linux-package-mixup.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+---
+title: Troubleshoot .NET Package Mix ups on Linux
+description: Learn about how to troubleshoot strange .NET package errors on Linux.
+author: omajid
+ms.date: 05/12/2021
+no-loc: ['usr','lib64','share','dotnet','libhostfxr.so', 'fxr', 'FrmaeworkList.xml', 'System.IO.FIleNoteFoundException']
```
Comment (omajid) [inline docs/core/install/linux-package-mixup.md:6]: Should be `File` and `Not`, right?

Comment (adegeo) [reply docs/core/install/linux-package-mixup.md:6]: Stupid keyboard.. yeah it was the keyboards problem.. not me.

**Comment by adegeo** [comment]
@omajid Do you want to remain author on this? If issues pop up on github you'll be pinged and responsible for addressing them, is that OK?

**Review by adegeo — APPROVED** [review]

**Comment by omajid** [comment]
> Do you want to remain author on this? If issues pop up on github you'll be pinged and responsible for addressing them, is that OK?

No strong preference either way. How do you folks normally handle outside (to the docs team) contributions?

I would be more than happy to be pinged if issues are reported in this doc, or someone if wants to make changes to it. I just can't promise a quick turnaround on the review/response.

File: docs/core/install/linux-package-mixup.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+- Generic errors without clear messages
+
+  - [Core #4605: cannot run "dotnet new console"](https://github.com/dotnet/core/issues/4605)
+  - [Core #4644: Cannot install .NET Core SDK 2.1 on Fedora 32](https://github.com/dotnet/core/issues/4655)
+  - [Runtime #49375: After updating to 5.0.200-1 using package manager, it appears that no sdks are installed](https://github.com/dotnet/runtime/issues/49375)
+
+## Next steps
+
+- [Install .NET on Linux](linux.md)
```
Comment (omajid) [inline docs/core/install/linux-package-mixup.md:133]: Heh, maybe installing .NET isn't the next step after troubleshooting the .NET installation? :smile:

Comment (adegeo) [reply docs/core/install/linux-package-mixup.md:133]: It's the "overview" and list of supported versions. I should probably change the title here...

File: docs/core/install/linux-package-mixup.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
+---
+title: Troubleshoot .NET Package Mix ups on Linux
+description: Learn about how to troubleshoot strange .NET package errors on Linux.
+author: omajid
+ms.date: 05/12/2021
+no-loc: ['usr','lib64','share','dotnet','libhostfxr.so', 'fxr', 'FrmaeworkList.xml', 'System.IO.FileNotFoundException']
```
Comment (omajid) [inline docs/core/install/linux-package-mixup.md:6]:
```suggestion
no-loc: ['usr','lib64','share','dotnet','libhostfxr.so', 'fxr', 'FrameworkList.xml', 'System.IO.FileNotFoundException']
```

**Pull request merged**